### PR TITLE
feat: Auth

### PR DESCRIPTION
**Pull request by dikshadhote**
This pull request contains Auth features:
1. Signup handler
2. Login handler
3. Logout handler
image
image
image

### CODE REVIEW

File: src/components/navbar/Navbar.js [hunk 1]
```diff
@@ -1,14 +1,24 @@
 import React from "react";
 import "./../../App.css";
-import { Link } from "react-router-dom";
+import { Link, useNavigate } from "react-router-dom";
 import storeLogo from "../../assets/location.png";
 import { useCart } from "../../context/cart-context";
 import { useWishlist } from "../../context/wishlist-context";
+import { useAuth } from "../../context/auth-context";
```
Comment (harshalini) [inline src/components/navbar/Navbar.js]: You can import useCart, useWishlist, useAuth in one file and then export from there

File: src/index.js [hunk 2]
```diff
@@ -7,18 +7,21 @@ import { BrowserRouter } from "react-router-dom";
 import { ProductsFilterProvider } from "./context/products-filter-context";
 import { CartProvider } from "./context/cart-context";
 import { WishlistProvider } from "./context/wishlist-context";
+import { AuthProvider } from "./context/auth-context";
```
Comment (harshalini) [inline src/index.js]: Same here

**Review by harshalini — APPROVED** [review]
Code is really well structured! Nicely done auth part!

**Review by Prabhav-Sharma — APPROVED** [review]
Clean and easy to navigate code. Good naming convention!
Suggestion: Can comment "I accept Terms and conditions" checkbox till you've added the T&C. passing down a suggestion I received by a mentor.

**Comment by dikshadhote** [comment]
> Clean and easy to navigate code. Good naming convention! Suggestion: Can comment "I accept Terms and conditions" checkbox till you've added the T&C. passing down a suggestion I received by a mentor.

Cool , will do that

**Pull request merged**